mlrun/projects/project.py

```diff
@@ -2884,6 +2885,23 @@ def _init_function_from_obj(func, project, name=None):
     return name or func.metadata.name, func
 
 
+def _mask_credentials_in_function_object(function):
+    for env in function.spec.env:
+        # if env contain a value and not value_from, this means it wasn't masked then we remove it
+        # if env contain a value_from, this means it was masked then no need to remove it
+        if env.name.startswith("V3IO_") and not env.value_from:
```

Comment:
Suggest we also check for existence of `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` which may be added as part of `mount_s3`.